torchgen/utils.py

```diff
@@ -106,7 +107,7 @@ def context(msg_fn: Callable[[], str]) -> Iterator[None]:
 # for getting mypy to do exhaustiveness checking
 # TODO: put this somewhere else, maybe
 def assert_never(x: NoReturn) -> NoReturn:
-    raise AssertionError(f"Unhandled type: {type(x).__name__}")
+    raise TypeError(f"Unhandled type: {type(x).__name__}")
```

Comment:
Why making this change?  Assertion error seems reasonable given the name function it's called from?

Reply:
Reverted.